rockcraft/models/project.py

```diff
@@ -331,18 +331,24 @@ def _check_unsupported_options(cls, values: Mapping[str, Any]) -> Mapping[str, A
 
         return values
 
-    @pydantic.validator("rock_license", always=True)
+    @pydantic.validator("license", always=True)
     @classmethod
-    def _validate_license(cls, rock_license: str) -> str:
+    def _validate_license(
+        cls, license: str | None  # pylint: disable=redefined-builtin
+    ) -> str | None:
         """Make sure the provided license is valid and in SPDX format."""
-        if rock_license == "proprietary":
+        if not license:
+            return None
+
+        if license == "proprietary":
             # This is the license name we use on our stores.
-            return rock_license
+            return license
 
-        lic: spdx_lookup.License | None = spdx_lookup.by_id(rock_license)  # type: ignore[reportUnknownMemberType]
+        lic: spdx_lookup.License | None = spdx_lookup.by_id(license)  # type: ignore[reportUnknownMemberType]
         if lic is None:
             raise CraftValidationError(
-                f"License {rock_license} not valid. It must be valid and in SPDX format."
+                f"License {license} not valid. It must be valid and in SPDX format.",
```

Comment:
It can also be `"proprietary"` apparently, the error message should point that out I believe